pkg/server/server.go

```
@@ -81,51 +156,110 @@ func (p PizzaOvenServer) handleRequest(w http.ResponseWriter, r *http.Request) {
+		http.Error(w, "Could not process input", http.StatusBadRequest)
+		return
+	}
+	cloneURLs, err := p.processOrg(data.Org, data.Archives)
+	if err != nil {
+		p.Logger.Errorf("Could not process org input: %v with error: %v", r.Body, err)
+		http.Error(w, "Could not process input", http.StatusInternalServerError)
+		return
+	}
 	if data.Wait {
-		err = p.processRepository(repoURLendpoint.String())
-		if err != nil {
-			p.Logger.Errorf("Could not process repository input: %v with error: %v", r.Body, err)
+		errorChannel := make(chan error)
+		wg := new(sync.WaitGroup)
+		wg.Add(len(cloneURLs))
+		for _, htmlURL := range cloneURLs {
+			go func(htmlURL string, wg *sync.WaitGroup, c chan error) {
+				_, repoURLendpoint, err := p.getValidRepoEndpoint(htmlURL)
+				if err != nil {
+					c <- err
+				} else {
+					c <- p.processRepository(repoURLendpoint)
+				}
+				defer wg.Done()
+			}(htmlURL, wg, errorChannel)
+		}
+		defer wg.Wait()
+		var errorFound = false
+		for err := range errorChannel {
+			if err != nil {
+				p.Logger.Error(err.Error())
+				errorFound = true
+			}
+		}
+		if errorFound {
 			http.Error(w, "Could not process input", http.StatusInternalServerError)
 			return
 		}
 	} else {
 		go func() {
-			err = p.processRepository(repoURLendpoint.String())
+			htmlURLs, err := p.processOrg(data.Org, data.Archives)
 			if err != nil {
-				p.Logger.Errorf("Could not process repository input: %v with error: %v", r.Body, err)
+				p.Logger.Errorf("Could not process org input: %v with error: %v", r.Body, err)
 				http.Error(w, "Could not process input", http.StatusInternalServerError)
 				return
 			}
+			errorChannel := make(chan error)
+
+			for _, htmlURL := range htmlURLs {
+				go func(htmlURL string, c chan error) {
+					_, repoURLendpoint, err := p.getValidRepoEndpoint(htmlURL)
+					if err != nil {
+						c <- err
+					} else {
+						c <- p.processRepository(repoURLendpoint)
+					}
+				}(htmlURL, errorChannel)
```

This should have a wait group or some way to block on all the go routines finishing. Otherwise, these will fire off and immediately get orphaned when the top level http request thread closes.

You'll also squash errors since you immediately go into checking the `errorChannel` without waiting for the work in the `processRepository` to potentially finish.